{
 "cells": [
  {
   "cell_type": "markdown",
   "id": "9428464e",
   "metadata": {},
   "source": [
    "# Sample Queries on the 1000 Genomes, gnomAD and ClinVar data lake\n",
    "\n",
    "In this notebook, we will demonstrate some sample genomics queries that are typically made by clinical geneticists and researchers on genomics variant data. We will use the parquet/ORC transformed variant data from the 3502 DRAGEN-reanalyzed 1000 Genomes dataset available in 3 different schemas at s3://1000genomes-DRAGEN-data-lake-ready/:\n",
    "\n",
    "1. var_partby_samples - Dataset partitioned by sample ID\n",
    "2. var_partby_chrom - Dataset partitioned by chromosome and bucketed by samples\n",
    "3. var_nested - Nested schema consisting of variant sites with sample IDs and genotypes that contain the variant\n",
    "\n",
    "We use the annotations from ClinVar that are available at https://registry.opendata.aws/clinvar/ to demonstrate how to make queries that use the raw variant data with annotations."
   ]
  },
  {
   "cell_type": "markdown",
   "id": "aaf12bc3",
   "metadata": {},
   "source": [
    "## Import Dependencies"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "99bfb6a6",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Requirement already satisfied: PyAthena in /home/ec2-user/anaconda3/envs/python3/lib/python3.6/site-packages (2.3.0)\n",
      "Requirement already satisfied: botocore>=1.5.52 in /home/ec2-user/anaconda3/envs/python3/lib/python3.6/site-packages (from PyAthena) (1.21.21)\n",
      "Requirement already satisfied: tenacity>=4.1.0 in /home/ec2-user/anaconda3/envs/python3/lib/python3.6/site-packages (from PyAthena) (8.0.0)\n",
      "Requirement already satisfied: boto3>=1.4.4 in /home/ec2-user/anaconda3/envs/python3/lib/python3.6/site-packages (from PyAthena) (1.18.21)\n",
      "Requirement already satisfied: s3transfer<0.6.0,>=0.5.0 in /home/ec2-user/anaconda3/envs/python3/lib/python3.6/site-packages (from boto3>=1.4.4->PyAthena) (0.5.0)\n",
      "Requirement already satisfied: jmespath<1.0.0,>=0.7.1 in /home/ec2-user/anaconda3/envs/python3/lib/python3.6/site-packages (from boto3>=1.4.4->PyAthena) (0.10.0)\n",
      "Requirement already satisfied: urllib3<1.27,>=1.25.4 in /home/ec2-user/anaconda3/envs/python3/lib/python3.6/site-packages (from botocore>=1.5.52->PyAthena) (1.26.6)\n",
      "Requirement already satisfied: python-dateutil<3.0.0,>=2.1 in /home/ec2-user/anaconda3/envs/python3/lib/python3.6/site-packages (from botocore>=1.5.52->PyAthena) (2.8.1)\n",
      "Requirement already satisfied: six>=1.5 in /home/ec2-user/anaconda3/envs/python3/lib/python3.6/site-packages (from python-dateutil<3.0.0,>=2.1->botocore>=1.5.52->PyAthena) (1.15.0)\n",
      "\u001b[33mWARNING: You are using pip version 21.1.3; however, version 21.2.4 is available.\n",
      "You should consider upgrading via the '/home/ec2-user/anaconda3/envs/python3/bin/python -m pip install --upgrade pip' command.\u001b[0m\n"
     ]
    }
   ],
   "source": [
    "import boto3, os\n",
    "\n",
    "s3 = boto3.resource('s3')\n",
    "glue = boto3.client('glue')\n",
    "cfn = boto3.client('cloudformation')\n",
    "import sys\n",
    "!{sys.executable} -m pip install PyAthena\n",
    "from pyathena import connect\n",
    "import pandas as pd\n",
    "from pyathena.pandas.util import as_pandas"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "772dbfd6",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "us-east-1\n"
     ]
    }
   ],
   "source": [
    "session = boto3.session.Session()\n",
    "region = session.region_name\n",
    "print(region)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 28,
   "id": "5d5f90a0",
   "metadata": {},
   "outputs": [],
   "source": [
    "databasename=\"\\\"1kg_full\\\"\"\n",
    "partitioned_chr = \"\\\"var_sortby\\\"\"\n",
    "partitioned_samples = \"\\\"var_partby_samples\\\"\"\n",
    "nested = \"\\\"var_nested\\\"\""
   ]
  },
  {
   "cell_type": "markdown",
   "id": "e541b25f",
   "metadata": {},
   "source": [
    "### Connect to Athena"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 43,
   "id": "547dd797",
   "metadata": {},
   "outputs": [],
   "source": [
    "conn = connect(s3_staging_dir='s3://athena-query-results-ss',region_name=region) #replace with your own bucket name\n"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "8c7fe36a",
   "metadata": {},
   "source": [
    "#### Let us explore the schema of the tables under the 1000 genomes transformed dataset"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "f6a8a06d",
   "metadata": {},
   "source": [
    "Here is the flat table schema for the data sorted by sample ID"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 44,
   "id": "4d2941a5",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "<class 'pandas.core.frame.DataFrame'>\n",
      "RangeIndex: 10 entries, 0 to 9\n",
      "Data columns (total 41 columns):\n",
      " #   Column                         Non-Null Count  Dtype  \n",
      "---  ------                         --------------  -----  \n",
      " 0   variant_id                     10 non-null     object \n",
      " 1   chrom                          10 non-null     object \n",
      " 2   pos                            10 non-null     int64  \n",
      " 3   alleles                        10 non-null     object \n",
      " 4   rsid                           0 non-null      object \n",
      " 5   qual                           10 non-null     float64\n",
      " 6   filters                        10 non-null     object \n",
      " 7   info.ac                        10 non-null     object \n",
      " 8   info.af                        10 non-null     object \n",
      " 9   info.an                        10 non-null     int64  \n",
      " 10  info.db                        10 non-null     bool   \n",
      " 11  info.dp                        10 non-null     int64  \n",
      " 12  info.end                       0 non-null      object \n",
      " 13  info.fs                        10 non-null     float64\n",
      " 14  info.fractioninformativereads  10 non-null     float64\n",
      " 15  info.lod                       0 non-null      object \n",
      " 16  info.mq                        10 non-null     float64\n",
      " 17  info.mqranksum                 0 non-null      object \n",
      " 18  info.qd                        10 non-null     float64\n",
      " 19  info.r2_5p_bias                0 non-null      object \n",
      " 20  info.readposranksum            0 non-null      object \n",
      " 21  info.sor                       10 non-null     float64\n",
      " 22  ad                             10 non-null     object \n",
      " 23  af                             10 non-null     object \n",
      " 24  dp                             10 non-null     int64  \n",
      " 25  fir2                           10 non-null     object \n",
      " 26  f2r1                           10 non-null     object \n",
      " 27  gp                             10 non-null     object \n",
      " 28  gq                             10 non-null     int64  \n",
      " 29  gt.alleles                     10 non-null     object \n",
      " 30  gt.phased                      10 non-null     bool   \n",
      " 31  mb                             10 non-null     object \n",
      " 32  pl                             10 non-null     object \n",
      " 33  pri                            10 non-null     object \n",
      " 34  ps                             2 non-null      float64\n",
      " 35  sb                             10 non-null     object \n",
      " 36  sq                             0 non-null      object \n",
      " 37  sample_id                      10 non-null     object \n",
      " 38  ref                            10 non-null     object \n",
      " 39  alt                            10 non-null     object \n",
      " 40  partition_0                    10 non-null     object \n",
      "dtypes: bool(2), float64(7), int64(5), object(27)\n",
      "memory usage: 3.2+ KB\n"
     ]
    }
   ],
   "source": [
    "cursor.execute(\"SELECT * from \" + databasename + \".\" + partitioned_samples + \"limit 10\")\n",
    "df = as_pandas(cursor)\n",
    "df.info()"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "e2cb0f45",
   "metadata": {},
   "source": [
    "Here is the schema for the flat table partitioned by chromosome and bucketed by samples. As you can see, the schema is very similar to the one partitioned by samples, except that \"chrom\" is the field that the data is partitioned by"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 30,
   "id": "28c9cd92",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "<class 'pandas.core.frame.DataFrame'>\n",
      "RangeIndex: 10 entries, 0 to 9\n",
      "Data columns (total 40 columns):\n",
      " #   Column                         Non-Null Count  Dtype  \n",
      "---  ------                         --------------  -----  \n",
      " 0   variant_id                     10 non-null     object \n",
      " 1   pos                            10 non-null     int64  \n",
      " 2   ref                            10 non-null     object \n",
      " 3   alt                            10 non-null     object \n",
      " 4   sample_id                      10 non-null     object \n",
      " 5   alleles                        10 non-null     object \n",
      " 6   rsid                           0 non-null      object \n",
      " 7   qual                           10 non-null     float64\n",
      " 8   filters                        10 non-null     object \n",
      " 9   info.ac                        0 non-null      object \n",
      " 10  info.af                        0 non-null      object \n",
      " 11  info.an                        0 non-null      object \n",
      " 12  info.db                        10 non-null     bool   \n",
      " 13  info.dp                        10 non-null     int64  \n",
      " 14  info.end                       0 non-null      object \n",
      " 15  info.fs                        0 non-null      object \n",
      " 16  info.fractioninformativereads  10 non-null     float64\n",
      " 17  info.lod                       10 non-null     float64\n",
      " 18  info.mq                        10 non-null     float64\n",
      " 19  info.mqranksum                 2 non-null      float64\n",
      " 20  info.qd                        0 non-null      object \n",
      " 21  info.r2_5p_bias                0 non-null      object \n",
      " 22  info.readposranksum            2 non-null      float64\n",
      " 23  info.sor                       0 non-null      object \n",
      " 24  ad                             10 non-null     object \n",
      " 25  af                             10 non-null     object \n",
      " 26  dp                             10 non-null     int64  \n",
      " 27  fir2                           10 non-null     object \n",
      " 28  f2r1                           10 non-null     object \n",
      " 29  gp                             0 non-null      object \n",
      " 30  gq                             0 non-null      object \n",
      " 31  gt.alleles                     10 non-null     object \n",
      " 32  gt.phased                      10 non-null     bool   \n",
      " 33  mb                             10 non-null     object \n",
      " 34  pl                             0 non-null      object \n",
      " 35  pri                            0 non-null      object \n",
      " 36  ps                             7 non-null      float64\n",
      " 37  sb                             10 non-null     object \n",
      " 38  sq                             10 non-null     float64\n",
      " 39  chrom                          10 non-null     object \n",
      "dtypes: bool(2), float64(8), int64(3), object(27)\n",
      "memory usage: 3.1+ KB\n"
     ]
    }
   ],
   "source": [
    "cursor.execute(\"SELECT * from \" + databasename + \".\" + partitioned_chr + \"limit 10\")\n",
    "df = as_pandas(cursor)\n",
    "df.info()"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "c6c7356b",
   "metadata": {},
   "source": [
    "Here is the nested schema. In this schema, most of the FORMAT and INFO fields are not retained. Each row is a variant site with an array consisting of sample IDs and genotypes"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 31,
   "id": "5fef85e6",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "<class 'pandas.core.frame.DataFrame'>\n",
      "RangeIndex: 10 entries, 0 to 9\n",
      "Data columns (total 6 columns):\n",
      " #   Column      Non-Null Count  Dtype \n",
      "---  ------      --------------  ----- \n",
      " 0   variant_id  10 non-null     object\n",
      " 1   pos         10 non-null     int64 \n",
      " 2   ref         10 non-null     object\n",
      " 3   alt         10 non-null     object\n",
      " 4   samples     10 non-null     object\n",
      " 5   chrom       10 non-null     object\n",
      "dtypes: int64(1), object(5)\n",
      "memory usage: 608.0+ bytes\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>variant_id</th>\n",
       "      <th>pos</th>\n",
       "      <th>ref</th>\n",
       "      <th>alt</th>\n",
       "      <th>samples</th>\n",
       "      <th>chrom</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chrY:G:C:2949388</td>\n",
       "      <td>2949388</td>\n",
       "      <td>G</td>\n",
       "      <td>C</td>\n",
       "      <td>[{id=NA18504, gts=[1]}, {id=HG03812, gts=[1]},...</td>\n",
       "      <td>chrY</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chrY:A:T:4634780</td>\n",
       "      <td>4634780</td>\n",
       "      <td>A</td>\n",
       "      <td>T</td>\n",
       "      <td>[{id=NA18504, gts=[1]}, {id=HG01986, gts=[1]},...</td>\n",
       "      <td>chrY</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chrY:C:T:4758611</td>\n",
       "      <td>4758611</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>[{id=NA18504, gts=[1]}, {id=HG01986, gts=[1]},...</td>\n",
       "      <td>chrY</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chrY:C:G:4806400</td>\n",
       "      <td>4806400</td>\n",
       "      <td>C</td>\n",
       "      <td>G</td>\n",
       "      <td>[{id=NA18504, gts=[1]}, {id=HG01986, gts=[1]},...</td>\n",
       "      <td>chrY</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chrY:C:CGTATATATATGTGTATATATATACGTGTATATATACAT...</td>\n",
       "      <td>4896733</td>\n",
       "      <td>C</td>\n",
       "      <td>CGTATATATATGTGTATATATATACGTGTATATATACAT</td>\n",
       "      <td>[{id=NA18504, gts=[1]}, {id=HG01986, gts=[1]},...</td>\n",
       "      <td>chrY</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>chrY:C:CTA:5667108</td>\n",
       "      <td>5667108</td>\n",
       "      <td>C</td>\n",
       "      <td>CTA</td>\n",
       "      <td>[{id=NA18504, gts=[1]}, {id=HG01986, gts=[1]},...</td>\n",
       "      <td>chrY</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>chrY:T:G:6792036</td>\n",
       "      <td>6792036</td>\n",
       "      <td>T</td>\n",
       "      <td>G</td>\n",
       "      <td>[{id=NA18504, gts=[1]}, {id=HG01986, gts=[1]},...</td>\n",
       "      <td>chrY</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>chrY:T:TTG:7524147</td>\n",
       "      <td>7524147</td>\n",
       "      <td>T</td>\n",
       "      <td>TTG</td>\n",
       "      <td>[{id=NA18504, gts=[1]}, {id=HG01986, gts=[1]},...</td>\n",
       "      <td>chrY</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>chrY:T:TG:7664244</td>\n",
       "      <td>7664244</td>\n",
       "      <td>T</td>\n",
       "      <td>TG</td>\n",
       "      <td>[{id=NA18504, gts=[1]}, {id=HG01986, gts=[1]},...</td>\n",
       "      <td>chrY</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>chrY:T:TA:8067814</td>\n",
       "      <td>8067814</td>\n",
       "      <td>T</td>\n",
       "      <td>TA</td>\n",
       "      <td>[{id=NA18504, gts=[1]}, {id=HG01986, gts=[1]},...</td>\n",
       "      <td>chrY</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                          variant_id      pos ref  \\\n",
       "0                                   chrY:G:C:2949388  2949388   G   \n",
       "1                                   chrY:A:T:4634780  4634780   A   \n",
       "2                                   chrY:C:T:4758611  4758611   C   \n",
       "3                                   chrY:C:G:4806400  4806400   C   \n",
       "4  chrY:C:CGTATATATATGTGTATATATATACGTGTATATATACAT...  4896733   C   \n",
       "5                                 chrY:C:CTA:5667108  5667108   C   \n",
       "6                                   chrY:T:G:6792036  6792036   T   \n",
       "7                                 chrY:T:TTG:7524147  7524147   T   \n",
       "8                                  chrY:T:TG:7664244  7664244   T   \n",
       "9                                  chrY:T:TA:8067814  8067814   T   \n",
       "\n",
       "                                       alt  \\\n",
       "0                                        C   \n",
       "1                                        T   \n",
       "2                                        T   \n",
       "3                                        G   \n",
       "4  CGTATATATATGTGTATATATATACGTGTATATATACAT   \n",
       "5                                      CTA   \n",
       "6                                        G   \n",
       "7                                      TTG   \n",
       "8                                       TG   \n",
       "9                                       TA   \n",
       "\n",
       "                                             samples chrom  \n",
       "0  [{id=NA18504, gts=[1]}, {id=HG03812, gts=[1]},...  chrY  \n",
       "1  [{id=NA18504, gts=[1]}, {id=HG01986, gts=[1]},...  chrY  \n",
       "2  [{id=NA18504, gts=[1]}, {id=HG01986, gts=[1]},...  chrY  \n",
       "3  [{id=NA18504, gts=[1]}, {id=HG01986, gts=[1]},...  chrY  \n",
       "4  [{id=NA18504, gts=[1]}, {id=HG01986, gts=[1]},...  chrY  \n",
       "5  [{id=NA18504, gts=[1]}, {id=HG01986, gts=[1]},...  chrY  \n",
       "6  [{id=NA18504, gts=[1]}, {id=HG01986, gts=[1]},...  chrY  \n",
       "7  [{id=NA18504, gts=[1]}, {id=HG01986, gts=[1]},...  chrY  \n",
       "8  [{id=NA18504, gts=[1]}, {id=HG01986, gts=[1]},...  chrY  \n",
       "9  [{id=NA18504, gts=[1]}, {id=HG01986, gts=[1]},...  chrY  "
      ]
     },
     "execution_count": 31,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "cursor.execute(\"SELECT * from \" + databasename +\".\" + nested + \" limit 10\")\n",
    "df = as_pandas(cursor)\n",
    "df.info()\n",
    "df"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "b8fb1305",
   "metadata": {},
   "source": [
    "### Queries on variant data alone\n",
    "Here are some examples of queries on just the variant data with no annotations"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "79c83b3a",
   "metadata": {},
   "source": [
    "**Example 1:** Query for variants that are on a specific gene BRCA1 (chr17:43044295-43125364) in a specific sample HG02625.\n",
    "This query will run faster and will need to scan only data within the specific sample partition, so we will use the dataset that is partitioned by sample"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 35,
   "id": "041e944f",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>variant_id</th>\n",
       "      <th>chrom</th>\n",
       "      <th>pos</th>\n",
       "      <th>alleles</th>\n",
       "      <th>rsid</th>\n",
       "      <th>qual</th>\n",
       "      <th>filters</th>\n",
       "      <th>info.ac</th>\n",
       "      <th>info.af</th>\n",
       "      <th>info.an</th>\n",
       "      <th>...</th>\n",
       "      <th>mb</th>\n",
       "      <th>pl</th>\n",
       "      <th>pri</th>\n",
       "      <th>ps</th>\n",
       "      <th>sb</th>\n",
       "      <th>sq</th>\n",
       "      <th>sample_id</th>\n",
       "      <th>ref</th>\n",
       "      <th>alt</th>\n",
       "      <th>partition_0</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr17:G:A:43044391</td>\n",
       "      <td>chr17</td>\n",
       "      <td>43044391</td>\n",
       "      <td>[G, A]</td>\n",
       "      <td>None</td>\n",
       "      <td>45.73</td>\n",
       "      <td>[]</td>\n",
       "      <td>[1]</td>\n",
       "      <td>[0.5]</td>\n",
       "      <td>2</td>\n",
       "      <td>...</td>\n",
       "      <td>[12, 17, 9, 10]</td>\n",
       "      <td>[81, 0, 50]</td>\n",
       "      <td>[0.0, 34.77, 37.77]</td>\n",
       "      <td>NaN</td>\n",
       "      <td>[13, 16, 13, 6]</td>\n",
       "      <td>None</td>\n",
       "      <td>HG02625</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>HG02625</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr17:CTT:C:43044804</td>\n",
       "      <td>chr17</td>\n",
       "      <td>43044804</td>\n",
       "      <td>[CTT, C, CTTTT]</td>\n",
       "      <td>None</td>\n",
       "      <td>308.73</td>\n",
       "      <td>[]</td>\n",
       "      <td>[1, 1]</td>\n",
       "      <td>[0.5, 0.5]</td>\n",
       "      <td>2</td>\n",
       "      <td>...</td>\n",
       "      <td>[0, 0, 17, 10]</td>\n",
       "      <td>[317, 263, 52, 391, 0, 54]</td>\n",
       "      <td>[0.0, 4.0, 7.0, 4.0, 8.0, 7.0]</td>\n",
       "      <td>NaN</td>\n",
       "      <td>[0, 0, 17, 10]</td>\n",
       "      <td>None</td>\n",
       "      <td>HG02625</td>\n",
       "      <td>CTT</td>\n",
       "      <td>C</td>\n",
       "      <td>HG02625</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr17:C:A:43045257</td>\n",
       "      <td>chr17</td>\n",
       "      <td>43045257</td>\n",
       "      <td>[C, A]</td>\n",
       "      <td>None</td>\n",
       "      <td>49.94</td>\n",
       "      <td>[]</td>\n",
       "      <td>[1]</td>\n",
       "      <td>[0.5]</td>\n",
       "      <td>2</td>\n",
       "      <td>...</td>\n",
       "      <td>[9, 10, 11, 7]</td>\n",
       "      <td>[85, 0, 50]</td>\n",
       "      <td>[0.0, 34.77, 37.77]</td>\n",
       "      <td>NaN</td>\n",
       "      <td>[12, 7, 8, 10]</td>\n",
       "      <td>None</td>\n",
       "      <td>HG02625</td>\n",
       "      <td>C</td>\n",
       "      <td>A</td>\n",
       "      <td>HG02625</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr17:A:G:43046604</td>\n",
       "      <td>chr17</td>\n",
       "      <td>43046604</td>\n",
       "      <td>[A, G]</td>\n",
       "      <td>None</td>\n",
       "      <td>48.94</td>\n",
       "      <td>[]</td>\n",
       "      <td>[1]</td>\n",
       "      <td>[0.5]</td>\n",
       "      <td>2</td>\n",
       "      <td>...</td>\n",
       "      <td>[9, 8, 5, 8]</td>\n",
       "      <td>[84, 0, 50]</td>\n",
       "      <td>[0.0, 34.77, 37.77]</td>\n",
       "      <td>NaN</td>\n",
       "      <td>[11, 6, 9, 4]</td>\n",
       "      <td>None</td>\n",
       "      <td>HG02625</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>HG02625</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr17:C:CA:43046757</td>\n",
       "      <td>chr17</td>\n",
       "      <td>43046757</td>\n",
       "      <td>[C, CA]</td>\n",
       "      <td>None</td>\n",
       "      <td>50.00</td>\n",
       "      <td>[]</td>\n",
       "      <td>[1]</td>\n",
       "      <td>[0.5]</td>\n",
       "      <td>2</td>\n",
       "      <td>...</td>\n",
       "      <td>[8, 5, 9, 6]</td>\n",
       "      <td>[56, 0, 50]</td>\n",
       "      <td>[0.0, 6.0, 9.0]</td>\n",
       "      <td>NaN</td>\n",
       "      <td>[7, 6, 10, 5]</td>\n",
       "      <td>None</td>\n",
       "      <td>HG02625</td>\n",
       "      <td>C</td>\n",
       "      <td>CA</td>\n",
       "      <td>HG02625</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>206</th>\n",
       "      <td>chr17:GTT:G:43123349</td>\n",
       "      <td>chr17</td>\n",
       "      <td>43123349</td>\n",
       "      <td>[GTT, G]</td>\n",
       "      <td>None</td>\n",
       "      <td>50.00</td>\n",
       "      <td>[]</td>\n",
       "      <td>[1]</td>\n",
       "      <td>[0.5]</td>\n",
       "      <td>2</td>\n",
       "      <td>...</td>\n",
       "      <td>[3, 5, 7, 3]</td>\n",
       "      <td>[52, 0, 50]</td>\n",
       "      <td>[0.0, 2.0, 5.0]</td>\n",
       "      <td>NaN</td>\n",
       "      <td>[5, 3, 4, 6]</td>\n",
       "      <td>None</td>\n",
       "      <td>HG02625</td>\n",
       "      <td>GTT</td>\n",
       "      <td>G</td>\n",
       "      <td>HG02625</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>207</th>\n",
       "      <td>chr17:A:G:43123628</td>\n",
       "      <td>chr17</td>\n",
       "      <td>43123628</td>\n",
       "      <td>[A, G]</td>\n",
       "      <td>None</td>\n",
       "      <td>47.93</td>\n",
       "      <td>[]</td>\n",
       "      <td>[1]</td>\n",
       "      <td>[0.5]</td>\n",
       "      <td>2</td>\n",
       "      <td>...</td>\n",
       "      <td>[10, 12, 6, 10]</td>\n",
       "      <td>[83, 0, 50]</td>\n",
       "      <td>[0.0, 34.77, 37.77]</td>\n",
       "      <td>NaN</td>\n",
       "      <td>[12, 10, 8, 8]</td>\n",
       "      <td>None</td>\n",
       "      <td>HG02625</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>HG02625</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>208</th>\n",
       "      <td>chr17:A:G:43124230</td>\n",
       "      <td>chr17</td>\n",
       "      <td>43124230</td>\n",
       "      <td>[A, G]</td>\n",
       "      <td>None</td>\n",
       "      <td>50.00</td>\n",
       "      <td>[]</td>\n",
       "      <td>[1]</td>\n",
       "      <td>[0.5]</td>\n",
       "      <td>2</td>\n",
       "      <td>...</td>\n",
       "      <td>[9, 7, 6, 12]</td>\n",
       "      <td>[85, 0, 50]</td>\n",
       "      <td>[0.0, 34.77, 37.77]</td>\n",
       "      <td>NaN</td>\n",
       "      <td>[8, 8, 13, 5]</td>\n",
       "      <td>None</td>\n",
       "      <td>HG02625</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>HG02625</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>209</th>\n",
       "      <td>chr17:T:C:43124331</td>\n",
       "      <td>chr17</td>\n",
       "      <td>43124331</td>\n",
       "      <td>[T, C]</td>\n",
       "      <td>None</td>\n",
       "      <td>50.00</td>\n",
       "      <td>[]</td>\n",
       "      <td>[1]</td>\n",
       "      <td>[0.5]</td>\n",
       "      <td>2</td>\n",
       "      <td>...</td>\n",
       "      <td>[6, 5, 6, 6]</td>\n",
       "      <td>[85, 0, 50]</td>\n",
       "      <td>[0.0, 34.77, 37.77]</td>\n",
       "      <td>NaN</td>\n",
       "      <td>[10, 1, 8, 4]</td>\n",
       "      <td>None</td>\n",
       "      <td>HG02625</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>HG02625</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>210</th>\n",
       "      <td>chr17:G:C:43125170</td>\n",
       "      <td>chr17</td>\n",
       "      <td>43125170</td>\n",
       "      <td>[G, C]</td>\n",
       "      <td>None</td>\n",
       "      <td>110.19</td>\n",
       "      <td>[]</td>\n",
       "      <td>[2]</td>\n",
       "      <td>[1.0]</td>\n",
       "      <td>2</td>\n",
       "      <td>...</td>\n",
       "      <td>[0, 0, 13, 8]</td>\n",
       "      <td>[148, 63, 0]</td>\n",
       "      <td>[0.0, 34.77, 37.77]</td>\n",
       "      <td>NaN</td>\n",
       "      <td>[0, 0, 6, 15]</td>\n",
       "      <td>None</td>\n",
       "      <td>HG02625</td>\n",
       "      <td>G</td>\n",
       "      <td>C</td>\n",
       "      <td>HG02625</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>211 rows × 41 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "               variant_id  chrom       pos          alleles  rsid    qual  \\\n",
       "0      chr17:G:A:43044391  chr17  43044391           [G, A]  None   45.73   \n",
       "1    chr17:CTT:C:43044804  chr17  43044804  [CTT, C, CTTTT]  None  308.73   \n",
       "2      chr17:C:A:43045257  chr17  43045257           [C, A]  None   49.94   \n",
       "3      chr17:A:G:43046604  chr17  43046604           [A, G]  None   48.94   \n",
       "4     chr17:C:CA:43046757  chr17  43046757          [C, CA]  None   50.00   \n",
       "..                    ...    ...       ...              ...   ...     ...   \n",
       "206  chr17:GTT:G:43123349  chr17  43123349         [GTT, G]  None   50.00   \n",
       "207    chr17:A:G:43123628  chr17  43123628           [A, G]  None   47.93   \n",
       "208    chr17:A:G:43124230  chr17  43124230           [A, G]  None   50.00   \n",
       "209    chr17:T:C:43124331  chr17  43124331           [T, C]  None   50.00   \n",
       "210    chr17:G:C:43125170  chr17  43125170           [G, C]  None  110.19   \n",
       "\n",
       "    filters info.ac     info.af  info.an  ...               mb  \\\n",
       "0        []     [1]       [0.5]        2  ...  [12, 17, 9, 10]   \n",
       "1        []  [1, 1]  [0.5, 0.5]        2  ...   [0, 0, 17, 10]   \n",
       "2        []     [1]       [0.5]        2  ...   [9, 10, 11, 7]   \n",
       "3        []     [1]       [0.5]        2  ...     [9, 8, 5, 8]   \n",
       "4        []     [1]       [0.5]        2  ...     [8, 5, 9, 6]   \n",
       "..      ...     ...         ...      ...  ...              ...   \n",
       "206      []     [1]       [0.5]        2  ...     [3, 5, 7, 3]   \n",
       "207      []     [1]       [0.5]        2  ...  [10, 12, 6, 10]   \n",
       "208      []     [1]       [0.5]        2  ...    [9, 7, 6, 12]   \n",
       "209      []     [1]       [0.5]        2  ...     [6, 5, 6, 6]   \n",
       "210      []     [2]       [1.0]        2  ...    [0, 0, 13, 8]   \n",
       "\n",
       "                             pl                             pri  ps  \\\n",
       "0                   [81, 0, 50]             [0.0, 34.77, 37.77] NaN   \n",
       "1    [317, 263, 52, 391, 0, 54]  [0.0, 4.0, 7.0, 4.0, 8.0, 7.0] NaN   \n",
       "2                   [85, 0, 50]             [0.0, 34.77, 37.77] NaN   \n",
       "3                   [84, 0, 50]             [0.0, 34.77, 37.77] NaN   \n",
       "4                   [56, 0, 50]                 [0.0, 6.0, 9.0] NaN   \n",
       "..                          ...                             ...  ..   \n",
       "206                 [52, 0, 50]                 [0.0, 2.0, 5.0] NaN   \n",
       "207                 [83, 0, 50]             [0.0, 34.77, 37.77] NaN   \n",
       "208                 [85, 0, 50]             [0.0, 34.77, 37.77] NaN   \n",
       "209                 [85, 0, 50]             [0.0, 34.77, 37.77] NaN   \n",
       "210                [148, 63, 0]             [0.0, 34.77, 37.77] NaN   \n",
       "\n",
       "                  sb    sq  sample_id  ref  alt  partition_0  \n",
       "0    [13, 16, 13, 6]  None    HG02625    G    A      HG02625  \n",
       "1     [0, 0, 17, 10]  None    HG02625  CTT    C      HG02625  \n",
       "2     [12, 7, 8, 10]  None    HG02625    C    A      HG02625  \n",
       "3      [11, 6, 9, 4]  None    HG02625    A    G      HG02625  \n",
       "4      [7, 6, 10, 5]  None    HG02625    C   CA      HG02625  \n",
       "..               ...   ...        ...  ...  ...          ...  \n",
       "206     [5, 3, 4, 6]  None    HG02625  GTT    G      HG02625  \n",
       "207   [12, 10, 8, 8]  None    HG02625    A    G      HG02625  \n",
       "208    [8, 8, 13, 5]  None    HG02625    A    G      HG02625  \n",
       "209    [10, 1, 8, 4]  None    HG02625    T    C      HG02625  \n",
       "210    [0, 0, 6, 15]  None    HG02625    G    C      HG02625  \n",
       "\n",
       "[211 rows x 41 columns]"
      ]
     },
     "execution_count": 35,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "cursor.execute(\"SELECT * from\" + databasename + \".\" + partitioned_samples +\n",
    "  \"WHERE chrom='chr17'\" \n",
    "  \"AND pos BETWEEN 43044295 AND 43125364 \"\n",
    "  \"AND sample_id = 'HG02625' \")\n",
    "\n",
    "df=as_pandas(cursor)\n",
    "df"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "5ee454ef",
   "metadata": {},
   "source": [
    "**Example 2:** Query to find samples that have variants in a specific region, in this case, the BRCA1 gene (chr17:43044295-43125364). \n",
    "This query can be run on the data that is partitioned by chromosome "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "id": "2466a98b",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>sample_id</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>HG03376</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>HG02635</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>NA18995</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>HG02293</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>HG01260</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3197</th>\n",
       "      <td>NA19675</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3198</th>\n",
       "      <td>HG02681</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3199</th>\n",
       "      <td>HG03054</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3200</th>\n",
       "      <td>HG00308</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3201</th>\n",
       "      <td>NA20797</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>3202 rows × 1 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "     sample_id\n",
       "0      HG03376\n",
       "1      HG02635\n",
       "2      NA18995\n",
       "3      HG02293\n",
       "4      HG01260\n",
       "...        ...\n",
       "3197   NA19675\n",
       "3198   HG02681\n",
       "3199   HG03054\n",
       "3200   HG00308\n",
       "3201   NA20797\n",
       "\n",
       "[3202 rows x 1 columns]"
      ]
     },
     "execution_count": 6,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "cursor.execute(\"SELECT DISTINCT(sample_id) from\" + databasename + \".\" + var_sortby\\\" \"\n",
    "  \"where chrom='chr17'\" \n",
    "  \"and pos between 43044295 and 43125364 \")\n",
    "\n",
    "df=as_pandas(cursor)\n",
    "df"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "bb1a811e",
   "metadata": {},
   "source": [
    "**Example 2a:** Query 2 on the nested data"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 36,
   "id": "b826bc46",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>id</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>HG00096</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>HG00097</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>HG00099</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>HG00100</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>HG00101</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3197</th>\n",
       "      <td>NA21137</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3198</th>\n",
       "      <td>NA21141</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3199</th>\n",
       "      <td>NA21142</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3200</th>\n",
       "      <td>NA21143</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3201</th>\n",
       "      <td>NA21144</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>3202 rows × 1 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "           id\n",
       "0     HG00096\n",
       "1     HG00097\n",
       "2     HG00099\n",
       "3     HG00100\n",
       "4     HG00101\n",
       "...       ...\n",
       "3197  NA21137\n",
       "3198  NA21141\n",
       "3199  NA21142\n",
       "3200  NA21143\n",
       "3201  NA21144\n",
       "\n",
       "[3202 rows x 1 columns]"
      ]
     },
     "execution_count": 36,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "cursor.execute(\"SELECT DISTINCT(sample.id) from (\"\n",
    "  \"SELECT samples from\" + databasename +  \".\" + nested + \n",
    "  \"where chrom='chr1'\" \n",
    "  \"and pos between 9033567 and 9142000\"\n",
    "  \") as f,\" \n",
    "  \"unnest(f.samples) as s(sample)\"\n",
    "  \"order by 1\")\n",
    "df=as_pandas(cursor)\n",
    "df"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "ecd03b42",
   "metadata": {},
   "source": [
    "**Example 3:** Query to find only samples that have homozygous variants in a specific region, in this case, the BRCA1 gene (chr17:43044295-43125364). "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 37,
   "id": "7f1e00c3",
   "metadata": {},
   "outputs": [
    {
     "ename": "SyntaxError",
     "evalue": "invalid syntax (<ipython-input-37-66281c0b3a8e>, line 4)",
     "output_type": "error",
     "traceback": [
      "\u001b[0;36m  File \u001b[0;32m\"<ipython-input-37-66281c0b3a8e>\"\u001b[0;36m, line \u001b[0;32m4\u001b[0m\n\u001b[0;31m    \"and array_join(gt.alleles, '|') in ('0|1')\")            )\u001b[0m\n\u001b[0m                                                             ^\u001b[0m\n\u001b[0;31mSyntaxError\u001b[0m\u001b[0;31m:\u001b[0m invalid syntax\n"
     ]
    }
   ],
   "source": [
    "cursor.execute(\"SELECT DISTINCT(sample_id) from\" + databasename + \".\" + partitioned_chr +\n",
    "  \"where chrom='chr17'\" \n",
    "  \"and pos between 43044295 and 43125364 \"\n",
    "  \"and array_join(gt.alleles, '|') in ('0|1')\")            )\n",
    "\n",
    "df=as_pandas(cursor)\n",
    "df"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "08f06566",
   "metadata": {},
   "source": [
    "### Queries joining variant data with ClinVar annotation data"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "5971943a",
   "metadata": {},
   "source": [
    "**ClinVar has a number of tables, but the one that we will use here is the summary_variants table. This table has many fields, but we will only be using a subset of them. We also need to create a Variant ID that can be used to join with the raw variant tables, so to make this easier, we will create a \"view\" of the summary_variants table that we can use with subsequent queries**"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 38,
   "id": "9f86f3b1",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "<pyathena.cursor.Cursor at 0x7f83e95b19b0>"
      ]
     },
     "execution_count": 38,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "cursor.execute(\"create or replace view clinvar as \"\n",
    "       \"select concat('chr', chromosome, ':', referenceallelevcf, ':', alternateallelevcf, ':', cast(positionvcf as varchar)) as variant_id, \"\n",
    "       \"chromosome as chrom, \"\n",
    "       \"positionvcf as pos, \"\n",
    "       \"referenceallelevcf as ref, \"\n",
    "       \"alternateallelevcf as alt, \"\n",
    "       \"genesymbol as genename, \" \n",
    "       \"clinicalsignificance as clinicalsignificance, \"\n",
    "       \"numbersubmitters as num_submitters, \"\n",
    "       \"reviewstatus as reviewstatus, \"\n",
    "       \"split(phenotypeids, ',') as phenotypeids, \"\n",
    "       \"split(phenotypelist, ',') as phenotypelist \"\n",
    "       \"from \\\"clinvar_summary_variants_dl-awsroda\\\".\\\"variant_summary\\\" where assembly = 'GRCh38'\"\n",
    "        \"and referenceallelevcf <> 'na' and alternateallelevcf <> 'na'\")\n"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "7f397b19",
   "metadata": {},
   "source": [
    "**Example 4:** Modify query 3 to find the number of samples that have \"variants of Unknown Significance\" in the BRCA1 gene"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 39,
   "id": "494ddce3",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>_col0</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>1550</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   _col0\n",
       "0   1550"
      ]
     },
     "execution_count": 39,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "cursor.execute(\"SELECT count(DISTINCT(sample.id)) FROM (\"\n",
    "       \"SELECT samples FROM \" + databasename + \".\" + nested + \" AS v \" \n",
    "       \"JOIN clinvar a \"\n",
    "       \"ON v.variant_id = a.variant_id \"\n",
    "       \"WHERE a.clinicalsignificance = 'Uncertain significance' \"\n",
    "       \"AND v.chrom='chr17' \"\n",
    "       \"AND v.pos BETWEEN 43044295 AND 43125364 \"\n",
    "       \") AS f, \"\n",
    "       \"UNNEST(f.samples) AS s(sample) \")\n",
    "df=as_pandas(cursor)\n",
    "df"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "3bb8c6bc",
   "metadata": {},
   "source": [
    "**Example 4a:** To have a greater degree of confidence in the results, we want to only consider those entries in ClinVar that have more than 1 submission. This query will filter the results from Example 4 to only consider those that have > 1 submitter."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 40,
   "id": "71b90065",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>_col0</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>41</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   _col0\n",
       "0     41"
      ]
     },
     "execution_count": 40,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "cursor.execute(\"SELECT count(DISTINCT(sample.id)) from (\"\n",
    "       \"SELECT samples FROM\" + databasename + \".\" + nested + \" as v \" \n",
    "       \"JOIN clinvar a \"\n",
    "       \"ON v.variant_id = a.variant_id \"\n",
    "       \"WHERE a.clinicalsignificance = 'Uncertain significance' \"\n",
    "       \"AND v.chrom='chr17' \"\n",
    "       \"AND v.pos BETWEEN 43044295 AND 43125364 \"\n",
    "       \"AND a.num_submitters > 1 \"\n",
    "       \") AS f, \"\n",
    "       \"UNNEST(f.samples) AS s(sample) \")\n",
    "df=as_pandas(cursor)\n",
    "df"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "d73b6f82",
   "metadata": {},
   "source": [
    "**Example 5a:**  Find all variants in this dataset that have a pathogenic variant in the BRCA1 gene (using the gene annotation from ClinVar). Running the same query using the chromosome and position range for the BRCA1 gene is much faster due to the partitioning of the data."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 41,
   "id": "9c20e05f",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>_col0</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   _col0\n",
       "0      4"
      ]
     },
     "execution_count": 41,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "cursor.execute(\"SELECT COUNT(DISTINCT sample_id) FROM \" + databasename + \".\" + partitioned_chr + \" as v \" \n",
    "               \"JOIN clinvar AS a \" \n",
    "               \"ON v.variant_id = a.variant_id \" \n",
    "               \"WHERE a.genename='BRCA1' AND clinicalsignificance='Pathogenic' \")\n",
    "\n",
    "df=as_pandas(cursor)\n",
    "df"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "f5d9ba21",
   "metadata": {},
   "source": [
    "**Example 5b:** Run Query 5a using the nested dataset"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 42,
   "id": "402d1dc9",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>_col0</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   _col0\n",
       "0      4"
      ]
     },
     "execution_count": 42,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "cursor.execute(\"SELECT COUNT(DISTINCT sample.id) FROM \" \n",
    "               \"(SELECT samples from \" + databasename + \".\" + nested + \" AS v \" \n",
    "               \"JOIN clinvar AS a \" \n",
    "               \"ON v.variant_id = a.variant_id \" \n",
    "               \"WHERE a.genename='BRCA1' AND clinicalsignificance='Pathogenic') AS f, \"\n",
    "               \"UNNEST(f.samples) AS s(sample)\")\n",
    "\n",
    "df=as_pandas(cursor)\n",
    "df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "04eae61f",
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "conda_python3",
   "language": "python",
   "name": "conda_python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.6.13"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
